### ISSUE TITLE
JetBrains: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Right-hand side of 'instanceof' is not an object

### ISSUE THREAD

**Issue by jay-fibi** [sourcegraph/jetbrains#650]
### Installation Information

Plugin version: ```5.3.27-nightly```
IDE version: ```IC-233.14015.106```

### Describe the bug

**Steps to reproduce:**

1. Have Cody plugin installed in your JetBrains IntelliJ IDEA editor
2. User is already logged in to Cody with PRO subscription
3. Have a large repo added i.e. https://github.com/JetBrains/intellij-community
4. Click on "Reindex Local Project" button and verify indexing starts
5. Switch to another project and switch back again.
6. If the "Reindex Local Project" button is showing enabled then click on it.

**Observed behavior:**
IDE error occurred and got stacktrace from Cody.

Stacktrace:
```text
org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Right-hand side of 'instanceof' is not an object

TypeError: Right-hand side of 'instanceof' is not an object
    at IndexManager.forceRefreshIndex (/snapshot/dist/agent.js)
    at async Object.callback (/snapshot/dist/agent.js)
    at async /snapshot/dist/agent.js
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:209)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)

```

### Expected behavior

No error should occur.

**Comment by taylorsperry** [comment]
I'm marking this as P1 because it only impacts local users. 

@pkukielka My understanding of next steps are that we need to:
1. Fix the button that reindexes local files for Free/Pro users
2. Remove the button that reindexes local files for Enterprise users (since they don't use symf, correct?)

**Comment by pkukielka** [comment]
@taylorsperry 
1. Actually looking at the stack trace we will have to fix either cody or agent, problem appeared not in the JetBrains plugin itself. But yes, we should fix it either way.
2. They also can use syfm. There was discussion about it on the JetBrains chat a week ago or so, and conclusion was that if the syrmf is enabled by default (and it was for all accounts, in cody, not in jetbrains) then it's better to show it and allow enterprise users to also do proper actions with it. So as it is now is by design.

**Comment by taylorsperry** [comment]
@pkukielka got it, so we still need to do the "additional work" described here: https://sourcegraph.slack.com/archives/C059N5FRYG3/p1707232720274889?thread_ts=1707209039.635139&cid=C059N5FRYG3?

**Comment by dominiccooney** [comment]
- I can't reproduce this in 5.5.9
- The "reindex" button is removed for enterprise users in #1308 
- I don't see this bug being reported since 5.5.5
- We shipped #1308 in 5.5.6, so it has probably fixed this issue.
